client.go

```diff
@@ -581,3 +585,80 @@ func nonce() []byte {
 	}
 	return nonce
 }
+
+func (c *Client) GetOidcPollCode(ctx context.Context, logger logrus.FieldLogger) (string, error) {
```

Comment:
nit: i tend to prefer keeping the name similar to what the API calls the endpoint, since that way we have less names for concepts when discussing it.

`EndpointPreauth`? (or EndpointAuthPreauth, but that feels a bit silly.)